MEQP2/Sniffs/Classes/ConstructorOperationsSniff.md

```diff
@@ -0,0 +1,32 @@
+# Rule: Only DI operations are allowed in constructor
+## Background
+In Magento 2, the PHP constructor of a class is used to inject dependencies: Constructor-based Dependency Injection (DI).
+Because the constructor its purpose is to inject dependencies, other operations (or other logic) should not be part of the
+constructor.
+
+## Reasoning
+The constructor should only assign the values of constructor arguments to internal variables, like shown below:
+
+```php
+class Foo
+{
+    private $bar;
+
+    public function __construct(
+        Bar $bar
+    ) {
+        $this->bar = $bar;
+    }
+}
+```
+
+Any other logic than these dependency assignment operators should be excluded from the constructor. Having additional logic in
+the constructor makes it harder to debug and test the class.
+
+## How it works
+This rule checks the constructor code to see if any other operators than `a + b` is used. If so, a warning is generated.
+
+## How to fix
+Move other logic to another method. In some cases, this could be `init()` or `_construct()` method that is already created by
```

Comment:
I would discourage using `_construct()` although it's apparently not yet deprecated. It seems to me that this would be a way to cheat the rule, since models with a `_construct` method call it from their constructor, so in the end there is happening more than assignments during object instantiation and that's what should be prevented.

Reply:
Entirely agree. I was writing down this text and then thought "OMG, am I really recommending `_construct()`". I had a better thought about it. The doc of the rule should not start too many other discussions. So I've now rephrased it to simply recommend people to move the non-DI code away from the constructor. https://github.com/jissereitsma/extdn-phpcs/commit/1114442c31409c31885e6c7770b54b2424c3cd6d